### ISSUE TITLE
The result for category lamp is zero

### ISSUE THREAD

**Issue by DianJJ** [mlpc-ucsd/Uni-3D#3]
Hi, 
I am able to reproduce the results using the pretrained model, and I am also able to train the 2d segmentation model and reproduce the results. But when I try to train the model in 3d, the result for category lamp is zero, I re-downloaded the code and tried again but it still happened. Is there any possible reasons that could lead to this problem? I attached my log file as below.  

I hope you can give me some suggestion, thanks!

[log.txt](https://github.com/mlpc-ucsd/Uni-3D/files/14338112/log.txt)

**Comment by DianJJ** [comment]
Hi, I tried again, and I found out the problem is that there seems to be some mismatch in the label index?
![image](https://github.com/mlpc-ucsd/Uni-3D/assets/94151507/24e9c705-07c2-4dda-aa77-ba99e9d2b66f)

**Comment by zx1239856** [comment]
The KeyError in your screenshot shows 12 which is the ceiling class, and it is not expected on the 3D-FRONT dataset.

**Comment by zx1239856** [comment]
On the Lamp class, PRQ is about ~11.9 (single-scale), and ~13.5 (multi-scale)

**Comment by DianJJ** [comment]
> The KeyError in your screenshot shows 12 which is the ceiling class, and it is not expected on the 3D-FRONT dataset.

I redownload everything and did not change the code. Do you have any idea how to fix this? Thanks!

**Comment by DianJJ** [comment]
> On the Lamp class, PRQ is about ~11.9 (single-scale), and ~13.5 (multi-scale)

I can reproduce the results using the pre-trained model, and everything is fine. So I think the problem exists in the training process.

**Comment by DianJJ** [comment]
Some updates, I found out the ignore_labels was not used in PanopticReconstructionQuality, so I changed the line 124 from if prediction_instance_id not in matched_ids_prediction to if (prediction_instance_id not in matched_ids_prediction) and (prediction_semantic_label not in self.ignore_labels). It temporarily solved the KeyError.   But I still cannot figure out why lamp category is zero.......

**Comment by zx1239856** [comment]
PanopticReconstructionQuality was copied from https://github.com/xheon/panoptic-reconstruction and it is not supposed to be modified. 

On 3D-FRONT dataset, there is no ceiling class (whose index is 12). In the predictions, key `12` should never appear unless there is something wrong with the 2D model, as the predicted segment label comes from that.

I would recommend double check the outputs of the 2D model.

**Comment by DianJJ** [comment]
Hi, thanks for correcting me, I removed the modification of PanopticReconstructionQuality.
I tried directly using the pre-trained 2d model as the MODEL.WEGHTS, but the same issue still exists. When I evaluate the 2d model, the results seem normal and I can reproduce the results. I noticed the NUM_CLASSES is 13 in mask2former_R50_bs16_160k.yaml. Does it have something to do with this?

**Comment by zx1239856** [comment]
I pushed a new commit which should solve the KeyError issue.

For the PRQ on the lamp class, please try to train more iterations.

**Comment by DianJJ** [comment]
Thanks for the new commit, it fixed the keyerror issue
I am starting to retrain the model, and I'll give updates when I finish
